### PR TITLE
Update release process issue template

### PR DESCRIPTION
**Pull request by RDIL**
# PR Summary
Fix broken HTML comment, fix grammar, fix punctuation.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/Release_Process.md [hunk 1]
```diff
@@ -22,21 +20,21 @@ Please **only** use this template if you are a maintainer.
...
           scripts to use the new package names, links, and `metadata.json`.
-- [ ] Kick off Release pipeline
+- [ ] Kick off Release pipeline.
 - [ ] Create the release tag and push the tag to `PowerShell/PowerShell` repository.
 - [ ] For preview releases,
-  merge the release branch to Github `master` with a merge commit.
+  merge the release branch to GitHub `master` with a merge commit.
 - [ ] For non-preview releases,
-  make sure all changes are either already in master or have
+  make sure all changes are already in master.
```
Comment (TravisEz13) [inline .github/ISSUE_TEMPLATE/Release_Process.md]:
```suggestion
  make sure all changes are either already in master or have a PR open.
```

Comment (TravisEz13) [inline .github/ISSUE_TEMPLATE/Release_Process.md]:
```suggestion
  make sure all changes are either already in master or have a PR open.
```

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/Release_Process.md (hunk 1 above)
Comment (TravisEz13) [reply .github/ISSUE_TEMPLATE/Release_Process.md]: Sorry, I forgot to complete my sentence

Comment (RDIL) [reply .github/ISSUE_TEMPLATE/Release_Process.md]: Applied.  I was wondering what happened there 😕

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@TravisEz13 Can you update your review please.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**